pkg/inventory/inventory.go

```diff
@@ -248,7 +250,35 @@ func (inventory *Inventory) GetVmByKey(key string) (*runtimev1alpha1.VirtualMach
 }
 
 // WatchVms returns a Watch interface of vm cache.
-func (inventory *Inventory) WatchVms(ctx context.Context, key string, labelSelector labels.Selector,
+func (i *Inventory) WatchVms(ctx context.Context, key string, labelSelector labels.Selector,
 	fieldSelector fields.Selector) (watch.Interface, error) {
-	return inventory.vmStore.Watch(ctx, key, labelSelector, fieldSelector)
+	return i.vmStore.Watch(ctx, key, labelSelector, fieldSelector)
+}
+
+// compareVirtualMachineObjects compare if two virtual machine objects are the same. Return true if same.
+func (i Inventory) compareVirtualMachineObjects(cached, discovered runtimev1alpha1.VirtualMachineStatus) bool {
+	// 1. Check if objects are same.
+	if reflect.DeepEqual(cached, discovered) {
+		return true
+	}
+	// 2. Check if NetworkInterface field differ.
+	if reflect.DeepEqual(cached.NetworkInterfaces, discovered.NetworkInterfaces) {
+		return false
```

Comment:
shouldn't this return true if nics are same?

Reply:
Azure returns nic address not in particular order. So we sort them and then compare